src/analysis/locate.ml

```diff
@@ -310,6 +310,16 @@ let load_cmt ~config comp_unit ml_or_mli =
       Ok (source_file, cmt_infos)
   | None -> Error ()
 
+let find_declaration_uid ~env ~decl_uid path =
+  let rec non_alias_declaration_uid path =
+    let md = Env.find_module path env in
+    match md.md_type with
+    | Mty_ident _ | Mty_signature _ | Mty_functor _ | Mty_for_hole -> md.md_uid
+    | Mty_alias path -> non_alias_declaration_uid path
+  in
+  try non_alias_declaration_uid path
+with Not_found -> decl_uid
```

Comment:
IIUC with this implementation if we have a chain of aliases and we fail to resolve one of them at any step (could be the very last), we'll just jump to the initial declaration.

Perhaps it might be more useful to stop on the last step we managed to reach? Otherwise people would have to call `jump-to-decl` over and over again to reach that point...

Something like this:
```ocaml
  let rec non_alias_declaration_uid fallback_uid path =
    match Env.find_module path env with
    | { md_type = Mty_alias path; md_uid; _ } -> non_alias_declaration_uid md_uid path
    | { md_type = (Mty_ident _ | ...); md_uid; _ } -> md_uid
    | exception Not_found -> fallback_uid
```
?

Reply:
Done in 5a3d79172a99d731aa257d21ea66616bc2eb6e67